src/pybind/mgr/orchestrator/module.py

```diff
@@ -349,6 +349,7 @@ def _add_host(self,
             labels = labels[0].split(',')
 
         s = HostSpec(hostname=hostname, addr=addr, labels=labels, status=_status)
+        s.validate()
```

Comment:
This is causing a teuthology test failure because it changes what error code we get in https://github.com/ceph/ceph/blob/master/qa/tasks/mgr/test_orchestrator_cli.py#L169-L170. It's expecting errno.ENOENT but is getting errno.EINVAL due to the host validation failing on the name "raise_not_implemented". I think we probably need to modify that test to be something like

```
ret = self._orch_cmd_result("host", "add", "raise-not-implemented")
self.assertEqual(ret, errno.ENOENT)
ret = self._orch_cmd_result("host", "add", "bad_hostname")
self.assertEqual(ret, errno.EINVAL)
```

Actual test failure if you'd like to look http://pulpito.front.sepia.ceph.com/adking-2022-03-23_02:54:35-orch:cephadm-wip-adk-testing-2022-03-22-2000-distro-basic-smithi/6754374/

Reply:
Code has changed. Now validation is done in the cephadm module so these test cases shouldn't fail (as they will not hit the cephadm backend).